include/bluetooth/gatt.h

```diff
@@ -623,9 +623,19 @@ ssize_t bt_gatt_attr_read_chrc(struct bt_conn *conn,
 	BT_GATT_ATTRIBUTE(_uuid, _perm, _read, _write, _value)
 
 #if IS_ENABLED(CONFIG_BT_SETTINGS_CCC_LAZY_LOADING)
-	#define BT_GATT_CCC_MAX (CONFIG_BT_MAX_CONN)
+	#if defined(CONFIG_BT_MAX_CONN)
+		#define BT_GATT_CCC_MAX (CONFIG_BT_MAX_CONN)
+	#else
+		#define BT_GATT_CCC_MAX 0
+	#endif
 #else
-	#define BT_GATT_CCC_MAX (CONFIG_BT_MAX_PAIRED + CONFIG_BT_MAX_CONN)
+	#if defined(CONFIG_BT_MAX_PAIRED) && defined(CONFIG_BT_MAX_CONN)
+		#define BT_GATT_CCC_MAX (CONFIG_BT_MAX_PAIRED + CONFIG_BT_MAX_CONN)
+	#elif defined(CONFIG_BT_MAX_CONN)
+		#define BT_GATT_CCC_MAX (CONFIG_BT_MAX_CONN)
+	#else
+		#define BT_GATT_CCC_MAX 0
+	#endif
 #endif
```

Comment:
Hmm, I would suggest that we either change 
```
#if IS_ENABLED(CONFIG_BT_SETTINGS_CCC_LAZY_LOADING)
```
to 
```
#if IS_ENABLED(CONFIG_BT_SETTINGS_CCC_LAZY_LOADING) && IS_ENABLED(CONFIG_BT_CONN)
```

or make `CONFIG_BT_SETTINGS_CCC_LAZY_LOADING` `depend on` `CONFIG_BT_CONN` (perhaps together with `CONFIG_BT_SETTINGS_CCC_STORE_ON_WRITE`

Reply:
Added mentioned Kconfig dependencies.